### ISSUE TITLE
How would you implement focus? Is it something Guido should explicitly help with?

### ISSUE THREAD

**Issue by JobLeonard** [fjenett/Guido#8]
I've been puzzling over this. I'm only bothering with the mouse - no "TAB to switch focus" problems, although that might also be interesting to investigate. There's two ways to go with this:
- focus is lost if the user clicks _anywhere_ else. This one's pretty darn easy to implement of course - just add a focus bool, set it to true if a mouse click overlaps and to false if not.
- focus is lost _only_ if the user clicks on another item that can get focus. This is the one I've been puzzling over. Getting focus is easy, but how to transmit to other UI elements it has lost focus? I figured that one way would be if the Guido library allows you to to send a message to the Interactive object, and that the latter "broadcasts" that message to all other objects registered by triggering a messageReceived function (like it now triggers mouse events). Or perhaps an 'Interactive.grabFocus(this)' function, which triggers a 'lostFocus' function in all other objects? I don't know if this is the best approach though.

Thoughts?

**Comment by fjenett** [comment]
Good point. Need to research … thanks!

**Comment by JobLeonard** [comment]
BTW, does Guido also send key presses to registered objects, like it does with the mouse events? If so, the "disabled list" example could be updated to highlight this, no? (should this be a separate issue?)

**Comment by fjenett** [comment]
Nope, only mouse stuff. But keypressed might be interesting for fast navigation, that is true. Yes, separate issue please.

**Comment by JobLeonard** [comment]
Done:

https://github.com/fjenett/Guido/issues/9
